administrator/components/com_banners/src/Model/BannerModel.php

```diff
@@ -429,6 +459,11 @@ public function save($data)
             $data['state'] = 0;
         }
 
+        if ($input->get('task') == 'save2copy' || $input->get('task') == 'copy') {
+            $data['clicks']  = 0;
+            $data['impmade'] = 0;
+        }
```

Comment:
```suggestion
            $data['state']   = 0;
            $data['clicks']  = 0;
            $data['impmade'] = 0;
        } elseif ($input->get('task') == 'copy') {
            $data['clicks']  = 0;
            $data['impmade'] = 0;
        }
```
It seems odd to be to check for the "save2copy" task 2 times, in the "if" condition in line 444 above and then again here.

So I suggest to do it differently: Add code for the "save2copy" task to the first "if" and use an "elseif" for the "copy" task only.

To be honest: I don't really know if we need the check for the "copy" task here. If it is not needed, the complete "elseif" of my suggestion can be dropped.

**_It's just a suggestion, no change request._**

**_Attention:_** When accepting my suggestion you should also change the comment in line 443 above the "if" to eg "Alter the name for save as copy and reset state, clicks and impressions" or something like this.

Reply:
I would prefer to set the values only on one place and keep the code as is.

Reply:
Do we really know if it needs the check for the "copy" task?

Reply:
See issue #40262 - my interpretation is "yes".